### ISSUE TITLE
Optimize the CSV upload of sites and addresses

### ISSUE THREAD

**Issue by maximilianong** [eclipse-tractusx/sig-release#948]
# Overview

The portal provides the option to upload sites and addresses for a legal entity via CSV.

https://github.com/eclipse-tractusx/sig-release/issues/700

However, this page should be revised at certain points to make it easier for users to understand.

## What's the benefit?
Companies can easily provide their site and address data, required for various use cases, to the data space via CSV upload. This allows them to contribute their own data to the Golden Record process without having to be a sharing member.

## What are the Risks/Dependencies ?
BPDM team provided an API for this CSV upload.

### Current implementation
https://github.com/eclipse-tractusx/sig-release/issues/700

### Proposed improvements
- Status
- .. @cofinity-ux will add more points

### Contributor

- @manojava-gk 

### Committer

- Committer 1
- Committer 2

## User Stories

- Issue 1, linked to specific repository
- Issue 2, linked to another specific repository

## Acceptance Criteria

- [ ] Criteria 1
- [ ] Criteria 2
- [ ] Criteria 3

#### Steps

1. Do something
2. Click something
3. Add something

#### Expected Result

1. Expectation
2. Expectation
3. Expectation

## Architectural Relevance

The following items are ensured (answer: yes) after this issue is implemented:

- [ ] This feature aligns with our current architectural guidelines
  - **Data Sovereignty:** All data sharing activities across company boundaries follow the [Catena-X Regulatory Framework](https://catenax-ev.github.io/docs/next/regulatory-framework/governance-framework), in particular the [Data Exchange Governance](https://catenax-ev.github.io/docs/next/regulatory-framework/20000ft/data-exchange-governance), and the [Dataspace Protocol](https://docs.internationaldataspaces.org/dataspace-protocol/overview/readme) via a compliant Connector (like the [tractusx-edc](https://github.com/eclipse-tractusx/tractusx-edc) or similar, see [Connector KIT](https://eclipse-tractusx.github.io/docs-kits/next/category/connector-kit))
    - [ ] [CX-0010 Business Partner Number](https://catenax-ev.github.io/docs/next/standards/CX-0010-BusinessPartnerNumber)
    - [ ] [CX-0013 Identity of Member Companies](https://catenax-ev.github.io/docs/next/standards/CX-0013-IdentityOfMemberCompanies)
    - [ ] [CX-0018 Data Space Connectivity (EDC)](https://catenax-ev.github.io/docs/next/standards/CX-0018-DataspaceConnectivity)
    - [ ] [CX-0049 DID Document Schema](https://catenax-ev.github.io/docs/next/standards/CX-0049-DIDDocumentSchema)
    - [ ] [CX-0050 Framework Agreement Credential](https://catenax-ev.github.io/docs/next/standards/CX-0050-FrameworkAgreementCredential)
    - [ ] [CX-0149 Verified Company Identity](https://catenax-ev.github.io/docs/next/standards/CX-0149-Dataspaceidentityandidentification)
  - **Interoperability:** Digital Twins are used (compliant to the [Digital Twin KIT](https://eclipse-tractusx.github.io/docs-kits/next/category/digital-twin-kit) and the [Industry Core KIT](https://eclipse-tractusx.github.io/docs-kits/next/category/industry-core-kit))
    - [ ] [CX-0001 EDC Discovery API](https://catenax-ev.github.io/docs/next/standards/CX-0001-EDCDiscoveryAPI)
    - [ ] [CX-0002 Digital Twins in Catena-X](https://catenax-ev.github.io/docs/next/standards/CX-0002-DigitalTwinsInCatenaX)
    - [ ] [CX-0018 Data Space Connectivity (EDC)](https://catenax-ev.github.io/docs/next/standards/CX-0018-DataspaceConnectivity)
  - **Data Format:**
    - [ ] The data model is based on a [published Semantic Model](https://github.com/eclipse-tractusx/sldt-semantic-models)
- [ ] The impact on the overall system architecture has been assessed. The Feature does not require changes to the architecture or any existing standard? Please have a look here on the [overarching architecture](https://eclipse-tractusx.github.io/docs/tutorials/e2e/inform/architecture)
- [ ] Potential risks or conflicts with existing architecture has been assessed

**Justification:** _(Fill this out, if at least one of the checkboxes above cannot be ticked. Contact the Architecture Management Committee to get an approval for the justification)_

## Additional information

- [ ] I am aware that my request may not be developed if no developer can be found for it. I'll try to contribute a developer (bring your own developer)

**Comment by Cofinity-UX** [comment]
@maximilianong 
Proposal for basic functionality, usability and wording fixes (I could not access the latest version of the reference implementation, maybe some points are already updated)
Kindly add me as Contributor in this ticket, thank you!

### **General structure of page:**
- Implementation of a headline, subline, text module as introduction and guidance for the adding address section (separation from My Company section). Concept, hierarchy and benefit of BPNL/S/A need to be communicated to users.
- Order of buttons in table: Add site first, add address after (to reflect the structure of address level and order of table)
- indicator that row element of table is clickable needs to be implemented (cursor should change into pointy hand)
- If user wants to copy information out of the table (indicated by the text cursor), it should not open the detailed view
- "1 row selected" information on the bottom of page is redundant
- If the BPD logic suggests that addresses relate to sites/legal entity, this information should be displayed in the table. (Legal entity can be seen as "Main Site" and should be selectable for adding addresses to it)
- Add Address: selection of site should be indicated by a dropdown field. If no site is available, the user should be enabled to create one
- Country Code input field: too vage description, no validation (in Registration flow there is a selection of country code (based on country list))
- Check refresh token of page, this might clear user input data after certain time
- Where is BPNA or BPNS information shown in successfull added data?


**### UX writing updates:**
**Main page:**
- Title of navigation item + page headline "Company address data" (Company data is not accurate enough)
- Wording of buttons don't correspond with opened modal (add address --> create new address)
- Wording of table columns should align with standard wording (e.g.: location --> Address, Type --> BPN Type)
- Status label "ready" is wrongly attributed with the action & icon. it needs to be updated accordingly to the process step (in review, pending etc..)
- Status label "success" is not accurate enough, better: "validated" "verified"

**Company Data Detail Modal:**
- Helpertext behind (?) should provide valuable information
- Legal entity number = Business Partner Number (BPN)

**Add Address/Site Modals:**
- Helptext behind (?) should not repeat the input field label names – provision of actual detail description of required input
- Error texts in inputfields: generic repetition of inputfield label is no error text. Error text has to correspond with all validation types and possible errors. (Company data step in registration flow has validation of the inputfields implemented, why not here?)
- Country identifier needs to be changed to Company identifier (country is the wrong term)

cc: @MaximilianHauer @ybidois

**Comment by Cofinity-UX** [comment]
Hi @SujitMBRDI,
can you explain what processes are happening in the background for those two status labels (context: adding site data on Company Data page). Thank you!
<img width="265" alt="Bildschirmfoto 2024-11-05 um 15 23 06" src="https://github.com/user-attachments/assets/cc17ad36-60c4-4eee-bee3-3ce4c45dc1fa">
<img width="212" alt="Bildschirmfoto 2024-11-05 um 15 23 18" src="https://github.com/user-attachments/assets/15ddf228-1355-4a1b-87d9-0fd6bdfc19bc">

cc: @maximilianong

**Comment by SujitMBRDI** [comment]
> Hi @SujitMBRDI, can you explain what processes are happening in the background for the two status labels "pending" and "ready" (context: adding site data on Company Data page). Thank you! <img alt="Bildschirmfoto 2024-11-05 um 15 23 06" width="265" src="https://private-user-images.githubusercontent.com/147708418/383205517-cc17ad36-60c4-4eee-bee3-3ce4c45dc1fa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4NzE5NTYsIm5iZiI6MTczMDg3MTY1NiwicGF0aCI6Ii8xNDc3MDg0MTgvMzgzMjA1NTE3LWNjMTdhZDM2LTYwYzQtNGVlZS1iZWUzLTNjZTRjNDVkYzFmYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNlQwNTQwNTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZTc1YTI4M2M4ZDZmN2I0NGU5MmJlOWE0MjBhYzI5MzJjODU5ZjE2MDZlYTM2ZDU1M2QyMzcwNWQyYWNiMTJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pFc28ama_U91nocTVTAuA-EcUp3QEfaeYLWxtOpoDhI"> <img alt="Bildschirmfoto 2024-11-05 um 15 23 18" width="212" src="https://private-user-images.githubusercontent.com/147708418/383205530-15ddf228-1355-4a1b-87d9-0fd6bdfc19bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4NzE5NTYsIm5iZiI6MTczMDg3MTY1NiwicGF0aCI6Ii8xNDc3MDg0MTgvMzgzMjA1NTMwLTE1ZGRmMjI4LTEzNTUtNGExYi04N2Q5LTBmZDZiZGZjMTliYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNlQwNTQwNTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNjk1NjEyY2NiNzIzOTZjMGRkOTM2MjQ0YTY1N2E0NWZhMjQyN2MzZGIzMGUwMTlkNThjZjQyMDU3MGRiMGJhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.WeDRLnHVUEzUCC5q4g8JyfIMuaHVM0_YWZ3pB3vgLAc">
> 
> cc: @maximilianong

Hello @Cofinity-UX,

The logic behind these two sharing state is as below,

`Ready` - When a sharing member uploads new business partner input data, the sharing member might not want to share this data immediately to the golden record process. The sharing member may perform additional checks on analysis on it first, prepare it further, before deciding it is fit to be shared.

`Pending` - After a sharing member has updated the business partner input data, the BPDM Gate should request a cleaning of the input data at the orchestrator. The orchestrator will create a new cleaning task and return its unique identifier to the requesting Gate. A Gate should then save the task identifier for its business partner data and update its sharing state.

**Comment by Cofinity-UX** [comment]
Hi @SujitMBRDI, thank you for clarifying.
Regarding the Ready status: right now, if I'm adding a site (Entry DEV (24.08)), the status goes automatically to pending.
Where is the functionality to actively share the site data to the Golden Record process?

**Comment by SujitMBRDI** [comment]
> Hi @SujitMBRDI, thank you for clarifying. Regarding the Ready status: right now, if I'm adding a site (Entry DEV (24.08)), the status goes automatically to pending. Where is the functionality to actively share the site data to the Golden Record process?

Hello @Cofinity-UX,

The functionality is totally depending on configuration you made while deploying application.
Here is [link](https://github.com/eclipse-tractusx/bpdm/blob/5da5d96347acec2314ab603a0247e6dccbe5b6ae/bpdm-gate/src/main/resources/application.yml#L44) for you reference. 
So, you can choose configurations as per the need like if you don't want business partner input data to directly ready to be shared then assign `starts-as-ready` to `false` likewise  configurations can be changed.

**Comment by Cofinity-UX** [comment]
Hi @SujitMBRDI,
if a customer subscribes to an application or service, the expectation is that the functionality is ready to use. So what you are describing is a scenario where a technically skilled user sets up this configuration in the backend, did I get that correctly? There is neither the information beforehand that this configuration can/must be taken nor any UI to do this connected to the Company Data page in the Portal?

**Comment by SujitMBRDI** [comment]
> Hi @SujitMBRDI, if a customer subscribes to an application or service, the expectation is that the functionality is ready to use. So what you are describing is a scenario where a technically skilled user sets up this configuration in the backend, did I get that correctly? There is neither the information beforehand that this configuration can/must be taken nor any UI to do this connected to the Company Data page in the Portal?
> 
> @maximilianong how is this configuration planned to face a customer?

Hello @Cofinity-UX,

I am not saying configuration planned to face a customer, you asked me to provide process which are happening in background. So as explained, due to default configuration on backend which is `starts-as-ready` set to `true`. It results when you add site then the status goes to automatically to pending and after golden record process you will receive status as `Success` or `Error`.

**Comment by Cofinity-UX** [comment]
@SujitMBRDI 
From a UX point of view I need to understand both, background processes which then affect the user interaction on touchpoint level. Since the user won't do the BE configuration, we (as provider of the service) need to decide, which configuration will be set in place, right?

If "starts-as-ready" is set to true, which interaction of the user will then set it to "pending"?

**Comment by SujitMBRDI** [comment]
> @SujitMBRDI From a UX point of view I need to understand both, background processes which then affect the user interaction on touchpoint level. Since the user won't do the BE configuration, we (as provider of the service) need to decide, which configuration will be set in place, right?
> 
> If "starts-as-ready" is set to true, which interaction of the user will then set it to "pending"?

@Cofinity-UX
Yes, you are right.

**Comment by stephanbcbauer** [comment]
Some hints from Release Management (@ther3sa) and Tractus-X Project Lead (@stephanbcbauer)
- Status currently in Inbox. ⇾ Only features with status backlog are considered in open planning
- Please add contributor and committer

**Comment by Cofinity-UX** [comment]
Hi @SujitMBRDI,
to determine with configuration to take for our customers, I still need to know: 
If "starts-as-ready" is set to true, what (or which interaction of the user) needs to happen to set the status to "pending"?

**Comment by Cofinity-UX** [comment]
Totally agree @SujitMBRDI, a session is needed. I'll send you an invite for next week, anyone from your side who should join (@MaximilianHauer?)

**Comment by manojava-gk** [comment]
@Cofinity-UX add me to the list. would like to participate in the meeting

**Comment by stephanbcbauer** [comment]
@MaximilianHauer mentioned -> Review by committer could be the problem. A lot todo

**Comment by stephanbcbauer** [comment]
@evegufy mentioned to prioritize this feature as high (from portal side)

**Comment by evegufy** [comment]
Acknowledging the fact that there is still refinement needed, Portal Committers:
@oyo
@ntruchsess 
@Phil91 
@evegufy

**Comment by ClosedSourcerer** [comment]
As discussed during the portal open meeting, I created the following issues:

https://github.com/eclipse-tractusx/portal-frontend/issues/1404
https://github.com/eclipse-tractusx/portal-frontend/issues/1403

**Comment by jupakajo** [comment]
The feature is deprioritized for several reasons:
1. The design of a proper user experience for csv upload needs more refinement.

- What fields are mandatory
- How do we map customer files to the data model

2. more importantly, the question remains how to deal within relations in the csv file. The expert group is currently prioritizing the relations for site creation. Once there is a solution, we can consider csv upload again.

Bottom line: backlog is the correct status

**Comment by MaximilianHauer** [comment]
@stephanbcbauer  should we remove the milestone then ?

**Comment by stephanbcbauer** [comment]
Yes please remove the milestone and add the me prep label for 25.09 thx

**Comment by maximilianong** [comment]
Update from last open refinement meeting: 23rd of April 2025:
This issue is kind of **ON HOLD**.
We know its relevant.
@jupakajo stated: first it should be possible to create multiple legal entities via front end and link them to each other. And than they will look into how this can be integrated into an CSV upload aswell.

**Comment by stephanbcbauer** [comment]
during open planning 25.09 ⇾ Need to be prioritized .... to not forget it in the next release, I added the new prep label for 25.12

**Comment by stephanbcbauer** [comment]
Was presented during open planning 25.12 -> no one drives this issue right now
